### ISSUE TITLE
Release Build not working with Pods

### ISSUE THREAD

**Issue by MathiasTim** [apache/cordova-ios#628]
# Bug Report

### What is expected to happen?
An archive is built when using the following Command:
```
npx ionic cordova build ios --device --prod --release --verbose -- --codeSignIdentity="Company Name" --developmentTeam="TeamID" --packageType="enterprise" --provisioningProfile="Profile-ID"
```

### What does actually happen?
I get the following error:
```
ld: warning: directory not found for option '-L/Users/mathiasmaier/Library/Developer/Xcode/DerivedData/ProjectName-byrqwtsixkwszxbdmgdleenawibs/Build/Intermediates.noindex/ArchiveIntermediates/Project Name/BuildProductsPath/Release-iphoneos/GoogleToolboxForMac'
ld: warning: directory not found for option '-L/Users/mathiasmaier/Library/Developer/Xcode/DerivedData/ProjectName-byrqwtsixkwszxbdmgdleenawibs/Build/Intermediates.noindex/ArchiveIntermediates/Project Name/BuildProductsPath/Release-iphoneos/Protobuf'
ld: warning: directory not found for option '-L/Users/mathiasmaier/Library/Developer/Xcode/DerivedData/ProjectName-byrqwtsixkwszxbdmgdleenawibs/Build/Intermediates.noindex/ArchiveIntermediates/Project Name/BuildProductsPath/Release-iphoneos/nanopb'
ld: library not found for -lGoogleToolboxForMac
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** ARCHIVE FAILED **
```

## Information
Building a debug version for device or simulator just works fine. Therefore I assume there is something wrong with the release build.

The release build just worked when using the old version of cordova-ios (4.5.5)

The plugin which has the Pods dependency is the phonegap-push-plugin

I get the same warning when trying to build an "archive" via Xcode (not the cli)

### Command or Code
```
npx ionic cordova build ios --device --prod --release --verbose -- --codeSignIdentity="Company Name" --developmentTeam="TeamID" --packageType="enterprise" --provisioningProfile="Profile-ID"
```

### Environment, Platform, Device
Release Builds, iOS

### Version information
Cordova Platform: cordova-ios 5.0.1
Cordova CLI: 9.0.0
Xcode: 10.1, MacOS 10.14

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by janpio** [comment]
Is this reproducible in a new, plain Cordova app? A minimal reproduction repository would really help to debug and later fix this issue. More information on how to create one: https://github.com/apache/cordova-contribute/blob/master/create-reproduction.md

**Comment by MathiasTim** [comment]
Thanks for the quick response @janpio.
I created a plain Cordova app and tada... it worked there.

So it seemed to be a problem just with my project. After a lot of searching and comparing my project to the plain cordova project we found out, that the version of the  platform inside the generated `Podfile` was iOS 11, but we specified iOS 10 in the `config.xml`.
In the plain test project the version number inside die `Podfile` always matched the one in the `config.xml`.

I could not find out why it always puts iOS 11 in there after the prepare step. We assume some other plugin causes it. Although only the described push plugin uses Pods. 
The only solution was to drop iOS 10 support and use iOS 11 as deployment target.

BTW the described error above was not the issue. We think the issues has to do with the different architecture requirements of iOS 10 and iOS 11.
```
Target 'Pods-Project Name' of project 'Pods' was rejected as an implicit dependency for 'libPods-ProjectName.a' because its architectures 'arm64' didn't contain all required architectures 'armv7 arm64'
```

Question is, is it possible to specify the platform ios version inside the generated `Podfile` ?
